File: src/Domain/Entities/ItemWeaponComponent.cs
namespace Crpg.Domain.Entities
{
    public class ItemWeaponComponent
    {
        public int Accuracy { get; set; }
        public int MissileSpeed { get; set; }
        public int StackAmount { get; set; }
        public int Length { get; set; }
        public int Handling { get; set; }
        public int BodyArmor { get; set; }
        public WeaponFlags Flags { get; set; }

        public int ThrustDamage { get; set; }
        public DamageType ThrustDamageType { get; set; }
        public int ThrustSpeed { get; set; }

        public int SwingDamage { get; set; }
        public DamageType SwingDamageType { get; set; }
        public int SwingSpeed { get; set; }
    }
}

File: src/WebUI/src/models/item-weapon-component.ts
import WeaponFlags from '@/models/weapon-flags';
import DamageType from '@/models/damage-type';
import WeaponClass from '@/models/weapon-class';

export default interface ItemWeaponComponent {
  class: WeaponClass;
  itemUsage: string;
  accuracy: number;
  missileSpeed: number;
  stackAmount: number;
  length: number;
  handling: number;
  bodyArmor: number;
  flags: WeaponFlags[];

  thrustDamage: number;
  thrustDamageType: DamageType;
  thrustSpeed: number;

  swingDamage: number;
  swingDamageType: DamageType;
  swingSpeed: number;
}
